### DISCUSSION TITLE
Version 0.30.4 uvicorn as Gunicorn worker with Nginx Docker Container Issue

### DISCUSSION THREAD

**Discussion by INNORAON** [Kludex/uvicorn#2407] [Q&A] [answered]
My Project is using these
- Python 3.12
- Django 4.2.13
- Gunicorn Latest
- Uvicorn Latest
- Nginx
With Docker Container

I have used uvicorn as gunicorn worker with Docker Container.
After Update 0.30.4 Uvicorn, some issue occured in gunicorn and nginx.
I could not connect my web project site.
WSGI Container was seem to normal status. Nginx Container was seem to normal status too.
But, when I connect my web site, nginx container return 504 Error ro 502 Error

So, I return uvicorn version 0.30.3. 
After that change, I could connect normally my site.
Check this Issue Please.

Thank a lot :)

**Comment by nlsfnr** [comment]
Can confirm this, no issues with either Nginx or the upstream application. No logs of failed requests showed up for uvicorn but nginx access / error logs showed timed-out requests.

**Downgrading to `uvicorn==0.30.3` fixed the issue**.

From /var/log/nginx/error.log (domain and IP addr. replaced):

`2024/08/01 12:04:20 [error] 824918#824918: *25 upstream timed out (110: Unknown error) while reading response header from upstream, client: 123.123.123.123, server: example.com, request: "GET / HTTP/1.1", upstream: "http://127.0.0.1:8000/", host: "example.com", referrer: "https://example.com"`

**Comment by razius** [comment]
We are having the same issue, with `0.30.4` our Kubernetes health checks started timing-out. 

We are using a Django application with gunicorn and the uvicorn worker.

- gunicorn==20.1.0
- Django==4.2.13

**Comment by Kludex** [comment, marked as answer]
This should be fixed on 0.30.5.
